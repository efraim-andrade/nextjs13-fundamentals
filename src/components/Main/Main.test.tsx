import { render, screen } from '@testing-library/react'

import { Main } from '.'

describe('<Main />', () => {
  it('should render correctly', () => {
    const { container } = render(<Main />)

    expect(screen.getByRole('heading', { name: /My main component/i })).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should be able to change the title', () => {
    render(<Main title="My new title" />)

    expect(screen.getByRole('heading', { name: /My new title/i })).toBeInTheDocument()
  })
})
