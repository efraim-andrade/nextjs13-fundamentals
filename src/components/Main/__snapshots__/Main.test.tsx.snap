// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render correctly 1`] = `
<main
  class="flex h-screen flex-col items-center justify-center"
>
  <h1
    class="text-3xl font-bold"
  >
    My main component
  </h1>
</main>
`;
